### ISSUE TITLE
Flycheck doesn't work on elm files in subdirectory

### ISSUE THREAD

**Issue by BrianHicks** [syl20bnr/spacemacs#3763]
This is a known issue (https://github.com/bsermons/flycheck-elm/issues/3) which follows from a known flycheck issue (https://github.com/flycheck/flycheck/issues/312)

The workaround that worked for me (provided by @bsermons) was to put the following in my `dotspacemacs/user-config`:

```
(add-hook 'elm-mode-hook (lambda ()
                             (setq default-directory
                                   (locate-dominating-file default-directory "elm-package.json"))))
```

I'd add it to the Elm layer, but I'm not sure where would be the most appropriate place. Not super familiar with how flycheck's hooks work. :\

**Comment by codedmart** [comment]
@BrianHicks that doesn't seem to work for me.

**Comment by BrianHicks** [comment]
One thing to try: as it is, the checker will create an `elm-package.json` in any directory with a `.elm` file. Those all have to be removed *before* the hook is run.

**Comment by codedmart** [comment]
@BrianHicks that seems to work. Is it possible to not have `elm-package.json` created in subdirectories?

**Comment by bsermons** [comment]
Here is a safer version in case the elm-package.json doesn't exist at all.  

Flycheck just sends the source file to elm-make and if there is no 'elm-package.json' in the cwd then it will create one.  Might be possible to get elm-make fix the issue by looking in the parent directories for a package file before trying to create one.

```elisp
  (add-hook 'elm-mode-hook (lambda ()
                             (setq default-directory
                                   (or (locate-dominating-file default-directory "elm-package.json")
                                       default-directory))))
```

**Comment by bharrisau** [comment]
The simplest place to add it is probably here:
https://github.com/syl20bnr/spacemacs/blob/master/layers/%2Blang/elm/packages.el#L28

Just need to make sure the hooks are run in the correct order. I've also suggested using the `elm-mode` function for finding the correct root.

    (add-hook 'elm-mode-hook (lambda ()
      (setq default-directory (elm--find-dependency-file-path)))

You'd also probably remove the lambda and define a function for it instead.

**Comment by syl20bnr** [comment]
@BrianHicks  What's the status about this ? Is this issue being PRed ?

**Comment by AlejandroCatalina** [comment]
I'm doing this, check PR in case it's wrong.

**Comment by d12frosted** [comment]
Fixed with released of Spacemacs v0.200. Let us know if you still have any problems with this issue.
